src/protocol.utility.ts

```diff
@@ -10,4 +10,23 @@ export class ProtocolUtility {
     public static async delay(ms: number): Promise<any> {
         return new Promise( resolve => setTimeout(resolve, ms) );
     }
+
+    /*
+        Given two dates, return difference in MS.
+        Assumes first input is newest, and oldest input is second resulting
+        in a positive return value.
+        eg:
+        ```
+                const startOf = new Date();
+                const deltaMS = timeDelta(new Date(), startOf));
+        ```
+     */
+    public static timeDelta(newest: Date, oldest: Date): number {
+        let result = newest.getTime() - oldest.getTime();
+        if (result <= 0) {
+            result = (result + 1000) % 1000;
```

Comment:
If result <=0 then the caller has put the values in the wrong order, right? Feels like an error, or maybe we want to return the absolute value. Not sure what the `result = (result + 1000) % 1000; ` is supposed to achieve?

Reply:
I dont really remember.  This logic is copied from the multiple places we use in it protocol and else where. https://github.com/kiva/protocol/blob/1156688e22bd185dd9d56e6cfdad1dc5d4aaa748/wallet_sync_service/src/sync/base-sync.service.ts#L185.  At this time, I would rather leave the logic in place so that we have parity with the code we are making common

Reply:
I counted at least 5 places this logic is used just in protocol.  It is also used in our aries projects.  It needs to be a common method.

Reply:
Thanks Matt, I’m movable on this. It seems weird though so let’s create another ticket to review / document / understand it and maybe make a unit test that proves the edge case that’s being catered to